client/apollo/react/src/Pagination/__tests__/PaginationCommon.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { render, screen } from "@testing-library/react";
+import { describe, it, expect, vi } from "vitest";
+import userEvent from "@testing-library/user-event";
+import { PaginationCommon, getItems } from "../PaginationCommon";
+import { ItemPagination } from "../ItemPagination/ItemPaginationApollo";
+import { ELLIPSIS } from "../ItemPagination/ItemPaginationCommon";
+
+describe("getItems", () => {
+  it("should return all items when numberPages <= 7", () => {
+    const result = getItems({
+      numberPages: 5,
+      currentPage: 3,
+      onChangePage: vi.fn(),
+    });
+
+    expect(result).toHaveLength(5);
+    expect(result[2].isCurrentPage).toBe(true);
+  });
+
+  it("should add ellipsis at end when currentPage < 6", () => {
+    const result = getItems({
+      numberPages: 10,
+      currentPage: 3,
+      onChangePage: vi.fn(),
+    });
+
+    expect(result).toHaveLength(8);
+    expect(result[6].page).toBe(ELLIPSIS);
+    expect(result[7].page).toBe(10);
+  });
+
+  it("should add ellipsis at start when currentPage is near end", () => {
+    const result = getItems({
+      numberPages: 10,
+      currentPage: 8,
+      onChangePage: vi.fn(),
+    });
+
+    expect(result[0].page).toBe(1);
+    expect(result[1].page).toBe(ELLIPSIS);
+  });
+
+  it("should add ellipsis at both ends for middle pages", () => {
+    const result = getItems({
+      numberPages: 15,
+      currentPage: 8,
+      onChangePage: vi.fn(),
+    });
+
+    expect(result[0].page).toBe(1);
+    expect(result[1].page).toBe(ELLIPSIS);
+    expect(result[5].page).toBe(ELLIPSIS);
+    expect(result[6].page).toBe(15);
+  });
+});
+
+describe("PaginationCommon", () => {
+  const defaultProps = {
+    numberPages: 5,
+    currentPage: 3,
+    onChangePage: vi.fn(),
+    ItemPaginationComponent: ItemPagination,
+  };
+
+  it("should render pagination with correct aria-label", () => {
+    render(<PaginationCommon {...defaultProps} />);
+    expect(screen.getByRole("navigation")).toHaveAttribute(
+      "aria-label",
+      "Pagination",
+    );
```

Comment:
Avec le getByRole tu peux tout faire d'un coup normalement : 

```suggestion
    expect(screen.getByRole("navigation", { name: Pagination })).toBeInTheDocument();
```